### ISSUE TITLE
Compatibility RepeaterMatrix

### ISSUE THREAD

**Issue by kixe** [thetuningspoon/AdminPageFieldEditLinks#5]
Currently the module is not compatible with ProField RepeaterMatrix. Would be nice to have the option to use Page Fields + AdminPageFieldEditLinks inside a RepeaterMatrix Field.

Good to know: PageTable are also not compatible with RepeaterMatrix.

**Comment by thetuningspoon** [comment]
Thanks for pointing this out. Will take a look.

**Comment by gingebaker** [comment]
Came across this problem also today. I think the main reason is that the JS and CSS files are not loaded when the inputfiled is loaded via ajax (repeaters or Inputfield::collapsedYesAjax).
( https://github.com/thetuningspoon/AdminPageFieldEditLinks/blob/master/AdminPageFieldEditLinks.module#L94 )

There is this Inputfield::renderReady function where all the core Inputfields add their JS and CSS now. 
( https://github.com/processwire/processwire/blob/master/wire/core/Inputfield.php#L1030 )

Unfortunately this renderReady is not hookable. However, if I make it hookable and test it, than ajax loading works just fine. The JS-Scripts are in the Head section. I made an feature request and asked in the forums:
https://github.com/processwire/processwire-requests/issues/12
https://processwire.com/talk/topic/14187-inputfieldrenderready-hook-alternative/

Also the AdminPageFieldEditLink.js needs some changes. Didn´t test too long but changing the ready function to 

``` javascript
$(document).ready(function() {
    AdminPageFieldEditLinks();
    $(document).on('reloaded', '.InputfieldPage', function(e) {
        AdminPageFieldEditLinks();
    });
});
```

...but I guess that would need some testing.

**Comment by thetuningspoon** [comment]
Thanks for the info, gingebaker. I wasn't sure how this was being handled for ajax fields. If you do get it working before I have a chance to look into it, feel free to submit a pull request.

**Comment by gingebaker** [comment]
I have a working solution here, but it does require making a core change (Inputfield::renderReady hookable).

My solution in the module looks like this:
There is a new hook for renderReady:
Note: I excluded the check for processes here.

``` php
$this->addHookAfter('InputfieldPage::renderReady', function($event) {
    $that = $event->object;
    if($that->editLinks || $that->newPageLink) {
       wire('modules')->get('JqueryUI')->use('modal');
       // load module scripts and styles for the list links
      $that->config->scripts->add($that->config->urls->AdminPageFieldEditLinks . "AdminPageFieldEditLinks.js?v=3");
      $that->config->styles->add($that->config->urls->AdminPageFieldEditLinks . "AdminPageFieldEditLinks.css?v=3");
      // Pass translatable strings to $config->js
      $that->config->js('AdminPageFieldEditLinks', array(
         'newPageLabel' => __('New'),
         'editPageLabel' => __('View'),
      ));
   }
});
```

The before Inputfield::render Hook is not needed anymore

The after Inputfield::render Hook stays the same. I also excluded the check for processes here. 

In the JS File there is this small change mentioned in the post above.

Than everythink works fine here. However, I think it does not make sense as long as the renderReady function is not hookable. Just to mention that it also works if I add the scripts/css and the js-config in the modules init method. But than everything gets loaded always wich is not cool...

Hope to hear some answers from the renderReady Request soon!

**Comment by thetuningspoon** [comment]
I had a chance to look at this more today and, as best as I can tell, it does seem that renderReady() is the only place that could be hooked, since this is the function called by the repeater in order to make sure all the necessary assets are pre-loaded. So I think you're correct. Hopefully Ryan can chime in on this on the forums.

**Comment by thetuningspoon** [comment]
Fixed in 3.1.1
